tests/test_multistore_file.py

```diff
@@ -0,0 +1,40 @@
+"""Unit tests for oauth2client.multistore_file."""
+
+import errno
+import os
+import tempfile
+import unittest
+
+from oauth2client import multistore_file
+
+
+class MultistoreFileTests(unittest.TestCase):
+
+    def test_lock_file_raises_ioerror(self):
+
+        class MockLockedFile(object):
+
+            def __init__(self, filename_str, error_code):
+                self.filename_str = filename_str
+                self.error_code = error_code
+
+            def open_and_lock(self):
+                raise IOError(self.error_code, '')
+
+            def is_locked(self):
+                return False
+
+            def filename(self):
+                return self.filename_str
+
+        filehandle, filename = tempfile.mkstemp()
+        os.close(filehandle)
+
+        try:
+            for error_code in (errno.EDEADLK, errno.ENOSYS, errno.ENOLCK):
+                multistore = multistore_file._MultiStore(filename)
+                multistore._file = MockLockedFile(filename, error_code)
+                # Should not raise even though the underlying file class did.
+                multistore._lock()
+        finally:
+            os.unlink(filename)
```

Comment:
Prefer this over `remove()`? Also, it's a `tempfile`, so we don't need to remove it, the extra I/O just slows down the test suite. Am I missing something?